shared/appengine.go

```diff
@@ -27,70 +27,79 @@ import (
-	// i.e. version-dot-wptdashboard{,-staging}.appspot.com
+	// GetVersionedHostname returns the AppEngine hostname for the current version,
+	// i.e. version-dot-wptdashboard{,-staging}.appspot.com.
 	GetVersionedHostname() string
+	// GetServiceHostname returns the AppEngine hostname for the given service (module) of the
+	// current version,
+	// i.e. version-dot-service-dot-wptdashboard{,-staging}.appspot.com.
+	// Note that if the current version does not have the specified service, AppEngine routing
+	// will fall back to its default version.
 	GetServiceHostname(service string) string
+
+	// GetResultsURL returns a URL for the wpt.fyi results page for the test runs loaded for the
+	// given filter.
 	GetResultsURL(filter TestRunFilter) *url.URL
+	// GetRunsURL returns a URL for the wpt.fyi results page for the test runs loaded for the
+	// given filter.
 	GetRunsURL(filter TestRunFilter) *url.URL
+	// GetResultsUploadURL returns a URL for uploading results (i.e. results receiver).
 	GetResultsUploadURL() *url.URL
+
+	// Simple wrapper of shared.IsFeatureEnabled (delegates to Datastore)
+	IsFeatureEnabled(featureName string) bool
+	GetUploader(uploader string) (Uploader, error)
 }
 
 // NewAppEngineAPI returns an AppEngineAPI for the given context.
-// Note that the context should be created using NewAppEngineContext
+// Note that the context should be created using NewAppEngineContext.
 func NewAppEngineAPI(ctx context.Context) AppEngineAPI {
-	return AppEngineAPIImpl{
+	return &appEngineAPIImpl{
 		ctx: ctx,
 	}
 }
 
-// AppEngineAPIImpl implements the AppEngineAPI interface.
-type AppEngineAPIImpl struct {
+// appEngineAPIImpl implements the AppEngineAPI interface.
+type appEngineAPIImpl struct {
 	ctx context.Context
-	// Cached client objects.
+	// Cached client objects
 	httpClient   *http.Client
 	githubClient *github.Client
 }
 
-// Context returns the context.Context for the API impl.
-func (a AppEngineAPIImpl) Context() context.Context {
+func (a appEngineAPIImpl) Context() context.Context {
 	return a.ctx
 }
 
-// GetHTTPClient returns an HTTP client in the current context.
-func (a AppEngineAPIImpl) GetHTTPClient() *http.Client {
+func (a *appEngineAPIImpl) GetHTTPClient() *http.Client {
 	if a.httpClient == nil {
 		a.httpClient = urlfetch.Client(a.ctx)
 	}
 	return a.httpClient
 }
 
-// GetSlowHTTPClient returns an HTTP client without timeout for the current
-// context.
-func (a AppEngineAPIImpl) GetSlowHTTPClient(timeout time.Duration) (*http.Client, context.CancelFunc) {
+func (a appEngineAPIImpl) GetSlowHTTPClient(timeout time.Duration) (*http.Client, context.CancelFunc) {
 	slowCtx, cancel := context.WithTimeout(a.ctx, timeout)
 	return urlfetch.Client(slowCtx), cancel
 }
 
-// GetGitHubClient returns a github client using the stored API token.
-func (a AppEngineAPIImpl) GetGitHubClient() (*github.Client, error) {
+func (a *appEngineAPIImpl) GetGitHubClient() (*github.Client, error) {
```

Comment:
what is the reasoning behind using *appEngineAPIImpl here? the only case I see is to help update githubClient but its a githubClient *github.Client